java/statestore/src/main/java/sleeper/statestore/s3/S3FileInfoStore.java

```diff
@@ -83,14 +75,12 @@ class S3FileInfoStore implements FileInfoStore {
     private S3FileInfoStore(Builder builder) {
         this.fs = Objects.requireNonNull(builder.fs, "fs must not be null");
         this.s3Path = Objects.requireNonNull(builder.s3Path, "s3Path must not be null");
-        this.dynamoRevisionIdTable = Objects.requireNonNull(builder.dynamoRevisionIdTable, "dynamoRevisionIdTable must not be null");
         this.rowKeyTypes = builder.rowKeyTypes;
         this.garbageCollectorDelayBeforeDeletionInMinutes = builder.garbageCollectorDelayBeforeDeletionInMinutes;
-        this.dynamoDB = Objects.requireNonNull(builder.dynamoDB, "dynamoDB must not be null");
         this.keySerDe = new KeySerDe(rowKeyTypes);
         this.fileSchema = initialiseFileInfoSchema();
         this.conf = builder.conf;
-        this.s3RevisionUtils = new S3RevisionUtils(dynamoDB, dynamoRevisionIdTable);
+        this.s3RevisionUtils = new S3RevisionUtils(builder.dynamoDB, builder.dynamoRevisionIdTable, builder.sleeperTable);
```

Comment:
sleeperTable is set on the builder and then used to create this S3RevisionUtils. The s3Path variable is created outside of this class and also contains sleeperTable. It might be cleaner to create s3Path from sleeperTable inside this constructor, as sleeperTable is effectively being passed in twice?